semver/is_semver.ts

```diff
@@ -25,22 +25,28 @@ export function isSemVer(value: unknown): value is SemVer {
   if (value === INVALID) return true;
   if (value === ANY) return true;
 
-  const { major, minor, patch, build = [], prerelease = [] } = value as Record<
-    string,
-    unknown
-  >;
-  const result = typeof major === "number" && isValidNumber(major) &&
+  const {
+    major,
+    minor,
+    patch,
+    build = [],
+    prerelease = [],
+  } = value as Record<string, unknown>;
+  return (
+    typeof major === "number" && isValidNumber(major) &&
     typeof minor === "number" && isValidNumber(minor) &&
     typeof patch === "number" && isValidNumber(patch) &&
     Array.isArray(prerelease) &&
+    prerelease
+      .every((v) => typeof v === "string" || typeof v === "number") &&
+    prerelease
+      .filter((v) => typeof v === "string")
+      .every((v) => isValidString(v)) &&
```

Comment:
```suggestion
      .filter((v) => isValidString(v)).length &&
```
Nit: Can this be done here and for the one below? Does it cater for edge cases?

Reply:
That wouldn't work, but it can be reduced to
```ts
prerelease.every((v) =>
  (typeof v === "string" && isValidString(v)) ||
  (typeof v === "number" && isValidNumber(v))
) 
```

Reply:
Actually:
```ts
prerelease.every((v) => isValidString(v)) || isValidNumber(v)) 
```